metis-harvesting/src/main/java/eu/europeana/metis/harvesting/http/HttpHarvesterImpl.java

```diff
@@ -62,13 +65,23 @@ public void harvestRecords(InputStream inputStream, CompressedFileExtension comp
         throw new HarvesterException("Problem saving archive.", e);
       }
 
+      AtomicInteger currentNumberOfIterations = new AtomicInteger();
+
       // Now perform the harvesting - go by each file.
       final HttpRecordIterator iterator = harvestRecords(tempFile);
       List<Pair<Path, Exception>> exception = new ArrayList<>(1);
       iterator.forEach(path -> {
         try (InputStream content = Files.newInputStream(path)) {
           action.accept(new ArchiveEntryImpl(path.getFileName().toString(),
-                  new ByteArrayInputStream(IOUtils.toByteArray(content))));
+              new ByteArrayInputStream(IOUtils.toByteArray(content))));
+          currentNumberOfIterations.getAndIncrement();
+          if (maxNumberOfIterations > 0) {
+            if (currentNumberOfIterations.get() < maxNumberOfIterations) {
+              return IterationResult.CONTINUE;
+            } else {
+              return IterationResult.TERMINATE;
+            }
+          }
```

Comment:
How about:
```suggestion
                    if (maxNumberOfIterations > 0 && currentNumberOfIterations.get() >= maxNumberOfIterations) {
                         return IterationResult.TERMINATE;
                    }
```
Instead?